compiler/noirc_frontend/src/hir_def/types/arithmetic.rs

```diff
@@ -212,4 +212,44 @@ impl Type {
             _ => None,
         }
     }
+
+    /// Try to unify equations like `(..) + 3 = (..) + 1`
+    /// by transforming them to `(..) + 2 =  (..)`
+    pub(super) fn try_unify_by_moving_constant_terms(
+        &self,
+        other: &Type,
+        bindings: &mut TypeBindings,
+    ) -> Result<(), UnificationError> {
+        if let Type::InfixExpr(lhs_a, op_a, rhs_a) = self {
+            if let Some(inverse) = op_a.inverse() {
+                if let Some(rhs_a) = rhs_a.evaluate_to_u32() {
```

Comment:
Could it be useful to attempt this in both directions? I.e. attempt to evaluate `lhs` and continue with `try_unify`?